edd-license-free-download.php

```diff
@@ -311,14 +293,12 @@ public static function comparison( $license_key, $download_id ) {
 		// products ids (that was saved in 'chosen' select box) the license will be checked against
 		$free_products = self::get_free_products_ids( $download_id );
 
-		// ids of products the license is for
-		$license_products = self::get_license_product_ids( $license_key );
+		// ids of product the license is for
+		$licensed_product = self::get_license_product_ids( $license_key );
 
 		// store the status of the license products. i.e whether they are among the products available for free or not
-		foreach ( $license_products as $product ) {
-			if ( in_array( $product, $free_products ) ) {
-				return true;
-			}
+		if ( in_array( $licensed_product, $free_products, true ) ) {

```

Comment:
Before we do this, we should make sure that all values are the same type. Since values can be filtered, we should make sure `$licensed_product` and all the values in `$free_products` are of the type `int`